File: src/main/java/com/ddubson/models/Book.java
package com.ddubson.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Table(name = "books")
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Book {
    @Id
    @GeneratedValue
    private Long id;

    private String title;

    @ManyToOne
    @JoinColumn(name = "author_id")
    private Author author;

    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinTable(name = "books_genres",
            joinColumns = {
                    @JoinColumn(name = "book_id", nullable = false, updatable = false)},
            inverseJoinColumns = {@JoinColumn(name = "genre_id",
                    nullable = false, updatable = false)})
    List<Genre> genres;
}

File: src/main/kotlin/com/ddubson/models/Book.kt
package com.ddubson.models

import java.io.Serializable
import javax.persistence.*

@Entity
@Table(name = "books")
data class Book(@Id @GeneratedValue
                val id: Long,
                val title: String,
                @ManyToOne(fetch = FetchType.LAZY, cascade = arrayOf(CascadeType.ALL))
                val author: Author,
                @ManyToMany(fetch = FetchType.LAZY, cascade = arrayOf(CascadeType.ALL))
                @JoinTable(name = "books_genres",
                        joinColumns = arrayOf(JoinColumn(name = "book_id", nullable = false, updatable = false)),
                        inverseJoinColumns = arrayOf(JoinColumn(name = "genre_id", nullable = false, updatable = false)))
                val genres: List<Genre>) : Serializable